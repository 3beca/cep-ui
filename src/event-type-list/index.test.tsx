import React from 'react';
import { render } from '@testing-library/react';
import EventTypeList from './index';

test('EventTypeList mount without crash', () => {
    const { getByText } = render(<EventTypeList />);
    const linkElement = getByText(/Listado de Reglas CEP/i);
    expect(linkElement).toBeInTheDocument();
});

test('EventTypeList snapshot', () => {
    const { container } = render(<EventTypeList />);
    expect(container).toMatchSnapshot();
});
