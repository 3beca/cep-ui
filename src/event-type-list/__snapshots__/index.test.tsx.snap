// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventTypeList snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-145"
  >
    Listado de Reglas CEP
    <div
      class="makeStyles-root-146"
    >
      <span>
        Table of Event Types
      </span>
      <div
        class="MuiTableContainer-root"
      >
        <table
          class="MuiTable-root"
        >
          <thead
            class="MuiTableHead-root makeStyles-head-147"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox"
                scope="col"
              >
                <span
                  aria-disabled="true"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-177 MuiCheckbox-root makeStyles-mainCheck-149 PrivateSwitchBase-disabled-179 Mui-disabled Mui-disabled Mui-disabled"
                  tabindex="-1"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      class="PrivateSwitchBase-input-180"
                      data-indeterminate="false"
                      disabled=""
                      type="checkbox"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                scope="col"
              >
                <p
                  class="MuiTypography-root makeStyles-headText-148 MuiTypography-body1"
                >
                  Event Type Name
                </p>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                <p
                  class="MuiTypography-root makeStyles-headText-148 MuiTypography-body1"
                >
                  Last updated
                </p>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                <p
                  class="MuiTypography-root makeStyles-headText-148 MuiTypography-body1"
                >
                  Created At
                </p>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-177 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      class="PrivateSwitchBase-input-180"
                      data-indeterminate="false"
                      type="checkbox"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
              >
                Dos
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2020-1-1 11:10:10
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2020-1-1 12:11:11
              </td>
            </tr>
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
                colspan="4"
                style="background-color: pink;"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
                >
                  Loading...
                </p>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
        >
          <div
            class="MuiTablePagination-spacer"
          />
          <p
            class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            Rows per page:
          </p>
          <div
            class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
          >
            <div
              aria-haspopup="listbox"
              aria-label="Rows per page:"
              class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
              role="button"
              tabindex="0"
            >
              10
            </div>
            <input
              type="hidden"
              value="10"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            1-10 of 10
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Previous page"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              title="Previous page"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Next page"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              title="Next page"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
